File: solutions/beecrowd/1240/1240.cpp
#include <cstdint>
#include <iostream>
#include <string>

int main() {
    std::int16_t n, i, la;
    std::string a, b;
    bool e;

    std::cin >> n;
    while (n--) {
        std::cin >> a >> b;

        e = true;

        if (b.length() > a.length()) {
            e = false;
        } else {
            la = a.length() - 1;

            for (i = b.length() - 1; i >= 0; i--) {
                if (b.at(i) != a.at(la)) {
                    e = false;
                    break;
                }
                la--;
            }
        }
        if (e) {
            std::cout << "encaixa" << std::endl;
        } else {
            std::cout << "nao encaixa" << std::endl;
        }
    }

    return 0;
}

File: solutions/beecrowd/1240/1240.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

n = int(input())

for _ in range(n):
    a, b = input().split()

    e = True

    if len(a) < len(b):
        e = False
    else:
        len_a = len(a) - 1

        for b_letter in b[::-1]:
            if b_letter != a[len_a]:
                e = False
                break
            len_a -= 1

    if e:
        output('encaixa\n')
    else:
        output('nao encaixa\n')
